tests/test_cli/test_generate_all_protocols.py

```diff
@@ -0,0 +1,267 @@
+    enforce(
+        return_code == 0,
+        f"Return code of {pprint.pformat(args)} is {return_code} != 0.",
+    )
+
+
+class TestGenerateAllProtcols(AEATestCaseMany):
+    """Test generate all protocols."""
+
+    use_packages_dir: bool = True
+    find_packages_patch: Any
+    package_path: Path
+
+    @classmethod
+    def setup_class(cls) -> None:
+        """Setup test class."""
+        super().setup_class()
+
+        cls.package_path = cls.t / "packages" / "fetchai" / "protocols" / "fipa"
+        cls.find_packages_patch = mock.patch(
+            "aea.cli.generate_all_protocols.find_protocols_in_local_registry",
+            return_value=(cls.package_path,),
+        )
+
+    def test_run(self, caplog) -> None:
+        """Test command invocation."""
+
+        with self.find_packages_patch, mock.patch.object(
+            AEAProject, "run_cli", new=_run_cli_patch
+        ), caplog.at_level(logging.INFO):
+            result = self.run_cli_command(
+                "generate-all-protocols",
+                str(self.packages_dir_path),
+                "--root-dir",
+                str(self.packages_dir_path.parent),
+            )
+
+            assert result.exit_code == 0
+            assert f"Processing protocol at path {self.package_path}"
+
+    def test_check_clean_pass(self) -> None:
+        """Test command invocation."""
+
+        with self.find_packages_patch, mock.patch.object(
+            AEAProject, "run_cli", new=_run_cli_patch
+        ), mock.patch(
+            "aea.cli.generate_all_protocols.check_working_tree_is_dirty",
+            return_value=True,
+        ):
+            result = self.run_cli_command(
+                "generate-all-protocols",
+                str(self.packages_dir_path),
+                "--root-dir",
+                str(self.packages_dir_path.parent),
+                "--check-clean",
+            )
+
+            assert result.exit_code == 0, result.output
+
+    def test_check_clean_fail(self) -> None:
+        """Test command invocation."""
+
+        with self.find_packages_patch, mock.patch.object(
+            AEAProject, "run_cli", new=_run_cli_patch
+        ), mock.patch(
+            "aea.cli.generate_all_protocols.check_working_tree_is_dirty",
+            return_value=False,
+        ):
+            with pytest.raises(AEATestingException, match="Exit code: 1"):
+                self.run_cli_command(
+                    "generate-all-protocols",
+                    str(self.packages_dir_path),
+                    "--root-dir",
+                    str(self.packages_dir_path.parent),
+                    "--check-clean",
+                )
+
+    def test_check_bump(self) -> None:
+        """Test command invocation."""
+
+        fipa_dir = self.t / "packages" / "fetchai" / "protocols" / "fipa"
+
+        def _download_patch(_: Any, temp_dir: str):
+            temp_fipa = Path(temp_dir, "fipa")
+            shutil.copytree(fipa_dir, temp_fipa)
```

Comment:
is this necessary? Since we already copy here
https://github.com/valory-xyz/open-aea/blob/146ee01cbb5d53b653d7a95c1224c93efc17575f/aea/test_tools/test_cases.py#L952-L955

Reply:
This is not same as the temporary package directory, the logic you're pointing to is a patch for the `download_package` method with downloads the package to the directory which was provided as the argument at the runtime.